moveit_core/robot_state/include/moveit/robot_state/cartesian_planner.h

```diff
@@ -0,0 +1,198 @@
+  }
+
+  /** \brief Compute the sequence of joint values that perform a general Cartesian path.
+
+     In contrast to the previous functions, the Cartesian path is specified as a set of \e waypoints to be sequentially
+     reached for the origin of a robot link (\e link). The waypoints are transforms given either in a global reference
+     frame or in the local reference frame of the link at the immediately preceeding waypoint. The link needs to move
+     in a straight line between two consecutive waypoints. All other comments apply. */
+  static double computeCartesianPath(
+      RobotState* start_state, const JointModelGroup* group, std::vector<std::shared_ptr<RobotState>>& traj,
+      const LinkModel* link, const EigenSTL::vector_Affine3d& waypoints, bool global_reference_frame,
+      const MaxEEFStep& max_step, const JumpThreshold& jump_threshold,
+      const GroupStateValidityCallbackFn& validCallback = GroupStateValidityCallbackFn(),
+      const kinematics::KinematicsQueryOptions& options = kinematics::KinematicsQueryOptions());
+
+  static double computeCartesianPath(
+      RobotState* start_state, const JointModelGroup* group, std::vector<std::shared_ptr<RobotState>>& traj,
+      const LinkModel* link, const EigenSTL::vector_Affine3d& waypoints, bool global_reference_frame, double max_step,
+      double jump_threshold_factor, const GroupStateValidityCallbackFn& validCallback = GroupStateValidityCallbackFn(),
+      const kinematics::KinematicsQueryOptions& options = kinematics::KinematicsQueryOptions())
+  {
+    return computeCartesianPath(start_state, group, traj, link, waypoints, global_reference_frame, MaxEEFStep(max_step),
+                                JumpThreshold(jump_threshold_factor), validCallback, options);
+  }
+
+  /** \brief Tests joint space jumps of a trajectory.
+
+     If \e jump_threshold_factor is non-zero, we test for relative jumps.
+     If \e revolute_jump_threshold  or \e prismatic_jump_threshold are non-zero, we test for absolute jumps.
+     Both tests can be combined. If all params are zero, jump detection is disabled.
+     For relative jump detection, the average joint-space distance between consecutive points in the trajectory is
+     computed. If any individual joint-space motion delta is larger then this average distance by a factor of
+     \e jump_threshold_factor, this step is considered a failure and the returned path is truncated up to just
+     before the jump.
+
+     @param group The joint model group of the robot state.
+     @param traj The trajectory that should be tested.
+     @param jump_threshold The struct holding jump thresholds to determine if a joint space jump has occurred.
+     @return The fraction of the trajectory that passed.
+  */
+  static double testJointSpaceJump(const JointModelGroup* group, std::vector<std::shared_ptr<RobotState>>& traj,
+                                   const JumpThreshold& jump_threshold);
```

Comment:
i'd prefer ``checkJointSpaceJump``, the word ``test`` has a distinct meaning in software. same for all other test functions here